addons/mail/static/tests/qunit_suite_tests/components/call_settings_menu_tests.js

```diff
@@ -0,0 +1,65 @@
+/** @odoo-module **/
+
+import {
+    start,
+    startServer,
+} from '@mail/../tests/helpers/test_utils';
+import { browser } from '@web/core/browser/browser';
+import { patchWithCleanup } from '@web/../tests/helpers/utils';
+
+QUnit.module('mail', {}, function () {
+QUnit.module('components', {}, function () {
+QUnit.module('call_settings_menu_tests.js');
+
+QUnit.test('Renders the call settings', async function (assert) {
+    assert.expect(4);
+
+    patchWithCleanup(browser, {
+        navigator: {
+            ...browser.navigator,
+            mediaDevices: {
+                enumerateDevices: () => Promise.resolve([
+                    { deviceId: 'mockAudioDeviceId', kind: 'audioinput', label: 'mockAudioDeviceLabel' },
+                    { deviceId: 'mockVideoDeviceId', kind: 'videoinput', label: 'mockVideoDeviceLabel' },
+                ]),
+            },
+        }
+    });
+
+    const pyEnv = await startServer();
+    const mailChannelId = pyEnv['mail.channel'].create({});
+    const { click, openDiscuss } = await start({
+        discuss: {
+            params: {
+                default_active_id: `mail.channel_${mailChannelId}`,
+            },
+        },
+    });
+    await openDiscuss();
+    await click('.o_ThreadViewTopbar_openCallSettingsButton');
+
+    assert.containsOnce(
+        document.body,
+        '.o_CallSettingsMenu',
+        "Should have a call settings menu"
+    );
+    assert.containsN(
+        document.body,
+        '.o_CallSettingsMenu_option',
+        5,
+        "Should have five options",
+    );
+    assert.containsOnce(
+        document.body,
+        'option[value=mockAudioDeviceId]',
+        "should have an option to select an audio input device",
+    );
+    assert.containsNone(
+        document.body,
+        'option[value=mockVideoDeviceId]',
+        "should not have an option to select a video input device",
+    );
```

Comment:
You just remind me I have a long awaiting task to turn `<select>` `<option>` to component. Because styling of `<select>` sucks, and template also sucks hard (it's alike OWL slots):

Task-2887231
https://github.com/odoo/odoo/pull/93904

We should eventually apply these changes in master